prow/cmd/mkbuild-cluster/main.go

```diff
@@ -199,14 +218,25 @@ func describeCluster(o options) (*describe, error) {
 	if d.Endpoint == "" {
 		return nil, errors.New("empty endpoint")
 	}
-	if len(d.Auth.ClusterCACertificate) == 0 {
-		return nil, errors.New("empty clusterCaCertificate")
-	}
 
-	if len(d.Auth.ClientKey) == 0 {
+	// Only generate new certificate if necessary
+	if o.getClientCert && (len(d.Auth.ClusterCACertificate) == 0 || len(d.Auth.ClientKey) == 0 || len(d.Auth.ClientCertificate) == 0) {
+		clientset, err := createDefaultClient()
+		if err != nil {
+			return nil, err
```

Comment:
`describeCluster()` can return a lot of different errors, please disambiguate and don't just return `err` from different functions.